components/lists/Friends.tsx

```diff
@@ -6,34 +6,39 @@ import api from "@/service"
 import { User } from "@/types"
 import { useStatus } from "@/contexts/StatusContext"
 import { useSignal } from "@/hooks/useSignal"
+import { FriendsSkeleton } from "@/components/cards/FriendsSkeleton"
+import { useQuery } from "@tanstack/react-query"
 
 export default function FriendsList() {
   const { friends, setFriends } = useStatus()
   const [friendsList, setFriendsList] = useState<User[]>([])
   const { signal } = useSignal()
 
   const fetchFriends = useCallback(async () => {
-    try {
-      const response = await api.get("/friends")
-      const allFriends = response.data.map((friend: User) => ({
-        id: friend.id,
-        names: friend.names,
-        username: friend.username || "",
-        email: friend.email,
-        profilePictureUrl: friend.profilePictureUrl || "",
-        selected: signal?.friends.some(
-          (signalFriend) => signalFriend.username === friend.username,
-        ),
-      }))
-      setFriendsList(allFriends)
-    } catch (error) {
-      console.error("Error fetching friends", error)
-    }
-  }, [signal?.friends])
+    const response = await api.get("/friends")
+    const allFriends = response.data.map((friend: User) => ({
+      id: friend.id,
+      names: friend.names,
+      username: friend.username || "",
+      email: friend.email,
+      profilePictureUrl: friend.profilePictureUrl || "",
+      selected: signal?.friends.some(
+        (signalFriend) => signalFriend.username === friend.username,
+      ),
+    }))
+    return allFriends
+  }, [signal])
+
+  const { data: friendsListData, isLoading } = useQuery({
+    queryKey: ["friends"],
+    queryFn: fetchFriends,
+  })
 
   useEffect(() => {
-    fetchFriends()
-  }, [fetchFriends])
+    if (friendsListData) {
+      setFriendsList(friendsListData)
+    }
+  }, [friendsListData])
```

Comment:
You could remove the `friendsList` state, altogether and use `friendsListData`, since it is not being manipulated anywhere